File: pset6/readability/readability.py
from cs50 import get_string


def count_letters(text):
    letters = 0
    n = len(text)
    for i in range(n):
        if text[i].isupper() or text[i].islower():
            letters += 1
    return letters


def count_words(text):
    words = 0
    n = len(text)
    for i in range(n):
        if text[i] == ' ':
            words += 1
    words += 1
    return words


def count_sentences(text):
    sentences = 0
    n = len(text)
    for i in range(n):
        if text[i] == '.' or text[i] == '?' or text[i] == '!':
            sentences += 1
    return sentences


text = get_string("Text: ")
letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)
mean_letters = letters / words * 100
mean_sentences = sentences / words * 100
coleman_liau_index = 0.0588 * mean_letters - 0.296 * mean_sentences - 15.8
index = round(coleman_liau_index)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string);
int count_words(string);
int count_sentences(string);

int main(void)
{
    string text = get_string("Text ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    double mean_letters = (double) letters / words * 100;
    double mean_sentences = (double) sentences / words * 100;
    double coleman_liau_index = 0.0588 * mean_letters - 0.296 * mean_sentences - 15.8;
    int index = round(coleman_liau_index);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    //add last word
    words++;

    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.')
        {
            sentences++;
        }
        else if (text[i] == '?')
        {
            sentences++;
        }
        else if (text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;

}